### PR TITLE
Share files from browser support

### LINKED ISSUE #1631: Allow link creation when sharing files
One should be able to create a sharing link to share a shared folder.

The current design in Figma [here](https://www.figma.com/file/wRMGqc11iQLVpDVmqXVXWK/ChainSafe-Files?node-id=4397%3A90289).

The first step would be to either select an existing folder, or to create a new shared folder. The sharing method would be the second step only in the case that we create a shared folder. 

So here are the flows:
- step1: Select either 1) an existing shared folder 2) create a new shared folder and give it a name
  if 1) then a click on "Share" would start the DL/encryption/UL
  if 2) then go to next step
- step2: We launch the DL/encryption/UL in the background and allow users to manage access with name/sharing link.

image

### PR DESCRIPTION
**Pull request by tanmoyAtb**
closes #1631

### CODE REVIEW

**Comment by asnaith** [comment]
Looking good for the most part on desktop and mobile. 

I found a small issue related to the refresh logic though and I think we need to trigger a refresh after a shared folder is deleted to ensure this correct modal is always displayed.

As demonstrated in the video below - after deleting all the shared folders the wrong modal was presented when trying to share again and it didn't update until the browser was refreshed. 

https://user-images.githubusercontent.com/3982665/142082112-b7f63f51-6c97-4575-bb74-b551c1683982.mov

**Comment by FSM1** [comment]
> I found a small issue related to the refresh logic though and I think we need to trigger a refresh after a shared folder is deleted to ensure this correct modal is always displayed.

This is caused by the bucket only being  marked as `status: deleting` after the request. We filter for these on the buckets overview, but not so in the `ShareModal.tsx` where `bucketOptions` is calculated. Just need to filter out any buckets with a `status: deleting` property.

File: packages/files-ui/src/Components/Modules/FileBrowsers/ShareModal.tsx [hunk 1]
```diff
@@ -121,82 +91,70 @@ const useStyles = makeStyles(
...
   const [destinationBucket, setDestinationBucket] = useState<BucketKeyPermission | undefined>()
+  const [isFolderCreationLoading, setIsFolderCreationLoading] = useState(false)
   const { buckets, transferFileBetweenBuckets } = useFiles()
   const { bucket, currentPath } = useFileBrowser()
   const { profile } = useUser()
   const [nameError, setNameError] = useState("")
   const inSharedBucket = useMemo(() => bucket?.type === "share", [bucket])
+
+  const classes = useStyles({
+    width: bucketToUpload ? 600 : 500
```
Comment (FSM1) [inline packages/files-ui/src/Components/Modules/FileBrowsers/ShareModal.tsx:155]: Any particular reason for this difference in width?

**Review by FSM1 — CHANGES_REQUESTED** [review]
image

Don't know if we want to deal with this in this ticket, but the modal covers the `cancel` button at the bottom of the screen now. We should also maybe consider renaming the button to `Close` as nothing is really being cancelled by closing the modal.

**Comment by tanmoyAtb** [comment]
> Looking good for the most part on desktop and mobile.
> 
> I found a small issue related to the refresh logic though and I think we need to trigger a refresh after a shared folder is deleted to ensure this correct modal is always displayed.
> 
> As demonstrated in the video below - after deleting all the shared folders the wrong modal was presented when trying to share again and it didn't update until the browser was refreshed.
> 
>  Screen.Recording.2021-11-16.at.3.17.26.PM.mov

@FSM1 thanks for the heads up, that status was it.
@asnaith should be good now. your observation is amazing !

File: packages/files-ui/src/Components/Modules/FileBrowsers/ShareModal.tsx (hunk 1 above)
Comment (tanmoyAtb) [reply packages/files-ui/src/Components/Modules/FileBrowsers/ShareModal.tsx:155]: Yeah so, The designs have this screen around 500px, however the manage shared folder modal (from shares overview) had this screen a bit larger. I wanted to keep the consistency.

**Comment by tanmoyAtb** [comment]
> image
> 
> Don't know if we want to deal with this in this ticket, but the modal covers the `cancel` button at the bottom of the screen now. We should also maybe consider renaming the button to `Close` as nothing is really being cancelled by closing the modal.

`Close` sounds more appropriate. 
As for the `close` button, I think we cover that in the modal redesign issue. #1729

**Review by asnaith — APPROVED** [review]
Nice fix @tanmoyAtb, looking good now after shared folder deletion, the right modal is shown.

**Review by FSM1 — APPROVED** [review]

**Pull request merged**